src/maniac_patch.cpp

```diff
@@ -600,4 +606,162 @@ bool ManiacPatch::GetKeyState(uint32_t key_id) {
 	}
 
 	return Input::IsRawKeyPressed(key);
+
+
+
+}
+
+bool ManiacPatch::CheckString(std::string str_l, std::string str_r, int op, bool ignore_case) {
+	std::regex specialChars { R"([-[\]{}()*+?.,\^$|#\s])" };
+	std::string sanitized_r = std::regex_replace(str_r, specialChars, R"(\$&)");
+	if (op > 1) { sanitized_r = ".*" + sanitized_r + ".*"; }
+
+	std::regex search;
+	if (ignore_case) { search = std::regex(sanitized_r, std::regex_constants::icase); }
+	else { search = std::regex(sanitized_r); }
+
+	switch (op) {
+	case 0: // eq
+	case 2: // contains (l contains r)
+		return std::regex_match(str_l, search);
+	case 1: // neq
+	case 3: // notContains (l does not contain r)
+		return !std::regex_match(str_l, search);
+	default:
+		return false;
```

Comment:
My review comment was lost here: Pls do not use ``std::regex`` for such comparisons as the implementing is slow and/or broken on some platforms.

use Utils::LowerCase or Utils::StrIcmp instead.

Reply:
done